### PR TITLE
Add doxygen in getfield.f, Part of #55

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
The doxygen has been added into the file.
It is valiaded at
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/getfield_8f.html#details

### CODE REVIEW

File: src/getfield.f [hunk 1]
```diff
@@ -1,121 +1,99 @@
+!>    @file
+!>    @brief This subroutine obtain grid information and the unpacked data
```
Comment (edwardhartnett) [inline src/getfield.f]: The brief statement applied to the whole file, not just the first subroutine. (In most of these files, there is only one subroutine, but in this file there is more than one.)

So it should be something like:

!> @brief Contains subroutines to get fields from...

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: src/getfield.f (hunk 1 above)
Comment (Hang-Lei-NOAA) [reply src/getfield.f]: You are right.

Comment (Hang-Lei-NOAA) [reply src/getfield.f]: The first subroutine is the main one. Following 4 are to unpack data from section 3-6.

**Review by GeorgeGayno-NOAA — APPROVED** [review]
Looks good.

File: src/getfield.f (hunk 1 above)
Comment (edwardhartnett) [reply src/getfield.f]: OK, put a summary of the file in the file @brief...

Comment (Hang-Lei-NOAA) [reply src/getfield.f]: Okay

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**